jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -32,11 +32,25 @@ def slice_indicator(self):
         self.figure.marks = self.figure.marks + slice_indicator.marks
         return slice_indicator
 
-    @property
+    @cached_property
     def slice_values(self):
+
         def _get_component(layer):
             try:
-                return layer.layer.get_component(self.slice_component_label).data
+                # Retrieve display units
+                display_spectral_units = self.jdaviz_app._get_display_unit('spectral')
```

Comment:
can `'spectral'` be accessed as a property (maybe `slice_display_unit_name` or something) so downstream can override when the x-axis isn't the spectral axis?   And maybe `display_spectral_units` can be `slice_display_units`?